pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentDictionaryCreator.java

```diff
@@ -211,6 +203,39 @@ public void build()
     }
   }
 
+  private void writeBytesValueDictionary(int numValues, byte[][] sortedByteArrays) throws IOException {
+
+    if (_useVarLengthDictionary) {
```

Comment:
Can this `if/else` be refactored such that the writer is set conditionally and the rest of the code is outside?

Reply:
Strictly speaking, `VarLengthBytesValueReaderWriter` and `FixedByteValueReaderWriter` are two implementations of readers but there is no common `Writer` interface here and hence the way the `byte[]` are written isn't uniform either.
Like I mentioned in another comment, I'm planning to do another PR to rename the classes and `ValueReader.java` interface so I'll see if this needs more cleanup then.